app/src/main/java/org/dhis2/usescases/development/ProgramRulesValidationsModule.kt

```diff
@@ -0,0 +1,79 @@
+package org.dhis2.usescases.development
+
+import android.content.Context
+import dagger.Module
+import dagger.Provides
+import org.dhis2.Bindings.valueTypeHintMap
+import org.dhis2.commons.di.dagger.PerActivity
+import org.dhis2.data.forms.dataentry.FormUiModelColorFactoryImpl
+import org.dhis2.data.forms.dataentry.FormView
+import org.dhis2.data.forms.dataentry.fields.FieldViewModelFactory
+import org.dhis2.data.forms.dataentry.fields.FieldViewModelFactoryImpl
+import org.dhis2.data.forms.dataentry.fields.LayoutProviderImpl
+import org.dhis2.form.data.FormRepositoryImpl
+import org.dhis2.form.model.coroutine.FormDispatcher
+import org.dhis2.form.ui.provider.DisplayNameProviderImpl
+import org.dhis2.form.ui.provider.HintProviderImpl
+import org.dhis2.form.ui.style.FormUiColorFactory
+import org.dhis2.form.ui.validation.FieldErrorMessageProvider
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.D2Manager
+
+@Module
+class ProgramRulesValidationsModule(val context: ProgramRulesValidationActivity) {
+
+    @Provides
+    @PerActivity
```

Comment:
I could be wrong here, but I thought that with the new version of dagger, there is no need to specify this type of annotation.

Reply:
They are not needed after the @Module tag. We can check, but that would be part of other ticket